### ISSUE TITLE
Datetime in incorrect format

### ISSUE THREAD

**Issue by mendoncaftw** [SapientGuardian/SapientGuardian.EntityFrameworkCore.MySql#41]
Hi, thanks for all the work btw.

I have a class:
```
    public class Subscription
    {
        public int Id { get; set; }

        //other props

        public DateTime CreatedAt { get; set; }

        public DateTime? EndsAt { get; set; }
    }
```
But when applying the migration I get the following error:

> MySql.Data.MySqlClient.MySqlException: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '/01/0001 00:00:00' at line 1

Here's the migration code:
```
        protected override void Up(MigrationBuilder migrationBuilder)
        {
            migrationBuilder.AddColumn<DateTime>(
                name: "CreatedAt",
                table: "Subscription",
                nullable: false,
                defaultValue: new DateTime(1, 1, 1, 0, 0, 0, 0, DateTimeKind.Unspecified));

            migrationBuilder.AddColumn<DateTime>(
                name: "EndsAt",
                table: "Subscription",
                nullable: true);
        }
```

Looking at the MySQL documentation and experimenting in the console I was able to reproduce the error like so:
> mysql> alter table shdev.plan add column plan.created_at DATETIME not null default 01/01/0001 00:00:00;
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '/01/0001 00:00:00' at line 1

Adding the **'** character around the datetime when executing the command correctly created a column but strangely it set the minimum date to the year 2001.

What should I do now? Is this from the EF Core lib? I could alter it through sql but the migrations would be in an inconsistent state.

Thanks in advance.

**Comment by SapientGuardian** [comment]
Per http://dev.mysql.com/doc/refman/5.7/en/datetime.html the minimum year you can store in MySql is the year 1000. It looks like you're trying to use year 1.

**Comment by mendoncaftw** [comment]
Hi @SapientGuardian, thanks for the quick response. Although that is correct, I'm pretty sure that's not the main problem. Just so I could test it out, I changed the migration to:
```
migrationBuilder.AddColumn<DateTime>(
                name: "CreatedAt",
                table: "Subscription",
                nullable: false,
                defaultValue: new DateTime(1970, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc));
```
and it still returned:

> You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '/01/1970 00:00:00' at line 1

I said initially I could reproduce this exact error message in SQL by omitting from the 01/01/1970 00:00:00 the surrounding ' . Therefore I'm inclined to think the SQL translation on the migration is **missing the single quote chararacters**. Am I going at this wrong?

Thanks for your help.

## **Update**
Tried with the following code before the _dotnet ef database update_:
```
migrationBuilder.AddColumn<DateTime>(
                name: "CreatedAt",
                table: "Subscription",
                nullable: false,
                defaultValueSql: "1970-01-01 00:00:00");
```
but it produced the same error.
Then tried with the following:
```
migrationBuilder.AddColumn<DateTime>(
                name: "CreatedAt",
                table: "Subscription",
                nullable: false,
                defaultValueSql: @"'1970-01-01 00:00:00'");
```
and it worked. 

So is this a problem with EF Core or your lib?

**Comment by SapientGuardian** [comment]
The problem is in this library, I think it's an easy fix. I'll try to get a patch out today or tomorrow. The bug is on https://github.com/SapientGuardian/SapientGuardian.EntityFrameworkCore.MySql/blob/master/src/SapientGuardian.EntityFrameworkCore.MySql/Migrations/MySQLMigrationsSqlGenerator.cs#L290 if you feel like having a go at it yourself.

**Comment by SapientGuardian** [comment]
The fix is in version 7.1.16, which will be published in a few minutes. Incidentally, MySql had no problem with your "year 1" date after I fixed the bug, so... yay documentation.
